File: python-project-euler-100/p006.py
from Solution import Solution

class P006(Solution):
    def solve(self):
        self.problem_number = 6
        last = 100
        sum = 0
        sum_of_squares = 0
    
        for i in range(1, last + 1):
            sum += i
            sum_of_squares += (i * i)
        
        result = (sum * sum) - sum_of_squares
        return result

def main():
	P006().run()

if __name__ == "__main__":
	main()

File: rust-project-euler-100/src/solutions/p006.rs
use crate::Solvable;

pub struct P006;
impl Solvable for P006 {
    fn problem_name(&self) -> &str { "p006" }
    fn solve(&self) -> i64 {
        let last = 100i64;
        let sum: i64 = (1..=last).sum();
        let sum_of_squares: i64 = (1..=last).map(|i| i * i).sum();
        sum * sum - sum_of_squares
    }
}
